articles/quickstart/spa/auth0-react/01.md

```diff
@@ -0,0 +1,299 @@
+
+As such, you can create an `Auth0ProviderWithHistory` function component that renders `Auth0Provider`. This wrapper component then calls `useHistory()` within its body to make the `history` object available within the scope of `Auth0Provider` and its  `onRedirectCallback` event handler.
+
+**Root component:**
+
+```javascript
+ReactDOM.render(
+  <BrowserRouter>
+    <Auth0ProviderWithHistory>
+      <App />
+    </Auth0ProviderWithHistory>
+  </BrowserRouter>,
+  document.getElementById("root")
+);
+```
+
+**`Auth0ProviderWithHistory` component:**
+
+```javascript
+const Auth0ProviderWithHistory = ({ children }) => {
+  const history = useHistory();
+
+  const onRedirectCallback = (appState) => {
+    history.push(appState?.returnTo || window.location.pathname);
+  };
+
+  return (
+    <Auth0Provider
+      domain="${account.namespace}"
+      client_id="${account.clientId}"
+      redirect_uri={window.location.origin}
+      onRedirectCallback={onRedirectCallback}
+    >
+      {children}
+    </Auth0Provider>
+  );
+};
+```
+
+The implementation of the authentication route guard may look like this `PrivateRoute` component:
+
+```javascript
+import { withLoginRequired } from "@auth0/auth0-react";
+
+const PrivateRoute = ({ component, ...args }) => (
+  <Route component={withLoginRequired(component)} {...args} />
+);
+```
+
+You can use the `PrivateRoute` component in the same way that you use the `Route` component from React Router:
+
+```javascript
+function App() {
+  const { isLoading } = useAuth0();
+
+  return (
+    <div className="App">
+      {isLoading ? (
+        <h2>Loading...</h2>
+      ) : (
+        <>
+          <NavBar />
+          <Switch>
+            <Route path="/" exact component={Home} />
+            <PrivateRoute path="/profile" component={Profile} />
+          </Switch>
+        </>
+      )}
+    </div>
+  );
+}
+```
+
+⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️
+
+**TODO: Final URL to sample.**
+ 
+⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️
+
+Take a look at this sample that puts this technique together while also implementing a navigation bar: "Using React Router with the Auth0 React SDK" [sample required]
+
+## Call an API
+
+Your React application may need to access protected data from an API that you are also [securing with Auth0](https://auth0.com/docs/microsites/protect-api/protect-api). For that use case, you'll need to retrieve an access token in the format of a [JSON Web Token (JWT)](https://jwt.io/) to authorize any calls you make to that secured API. 
+
+For the Auth0 React SDK to configure the access token correctly, you need to pass an `audience` property to the `Auth0Provider` as follows:
+
+```javascript
+import React from "react";
+import ReactDOM from "react-dom";
+import App from "./App";
+
+import { Auth0Provider } from "@auth0/auth0-react";
+
+ReactDOM.render(
+  <Auth0Provider
```

Comment:
If the user was following this from start to finish, they might find it confusing that they have to undo the stuff they did in `Auth0ProviderWithHistory` or adapt this to match it.

Reply:
Moved to CF.